src/winapi_support.rs

```diff
@@ -0,0 +1,32 @@
+use prelude::*;
+
+use BytesError;
+
+use winapi::shared::guiddef;
+
+impl Uuid {
+    /// Attempts to create a [`Uuid`] from a winapi `GUID`
+    ///
+    /// [`Uuid`]: ../struct.Uuid.html
+    fn from_guid(guid: guiddef::GUID) -> Result<Uuid, BytesError> {
+        Uuid::from_fields(
+            guid.Data1 as u32,
+            guid.Data2 as u16,
+            guid.Data3 as u16,
+            &(guid.Data4 as [u8; 8]),
+        )
+    }
+
+    /// Converts a [`Uuid`] into a winapi `GUID`
+    ///
+    /// [`Uuid`]: ../struct.Uuid.html
+    fn to_guid(&self) -> guiddef::GUID {
+        let (data1, data2, data3, data4) = self.to_fields_le();
```

Comment:
Document assumption that the `Uuid` is LE

Reply:
I could have it use whichever endianness the `Uuid` is. `Uuid::to_fields_le` is the only method I found, and is actually the outlier, with the other methods being prefixed `as_`